atlante/atlante.go

```diff
@@ -33,9 +33,12 @@ type ImgStruct struct {
 	intermediate bool
 	// Have we generated the file?
 	// This will allow us to generate the file only when requested
-	generated bool
-	lck       sync.Mutex
-	image     *image.Image
+	generated                 bool
+	lck                       sync.Mutex
+	image                     *image.Image
+	startGenerationCallBack   func()
```

Comment:
these are empty funcs so they can be any function. Maybe make the funcs a type and then set this property to the type?

Reply:
I debated about this. Do we really need a type for such a simple call back function?

Reply:
maybe not, and I guess if it's not accepting anything it's unnecessary.